sphinx_rtd_theme/static/js/versions.js_t

```diff
@@ -123,3 +124,68 @@ function renderLanguages(config) {
       document.dispatchEvent(event);
     });
  });
+{%- endif %}
+
+
+{%- if theme_version_selector or theme_language_selector %}
+function onSelectorSwitch(event) {
+     const option = event.target.selectedIndex;
+     const item = event.target.options[option];
+     window.location.href = item.dataset.url;
+ }
+
+document.addEventListener("readthedocs-addons-data-ready", function(event) {
+  const config = event.detail.data();
+
+  const dropdownSelectors = document.querySelector("div.dropdown-selectors");
+
+  {%- if theme_version_selector %}
+  const versionSelect = `
+    <select id="readthedocs-version-selector">
+      ${ config.versions.active.map(
+          (version) => `
+          <option
+              value="${ version.slug }"
+              ${ config.versions.current.slug === version.slug ? 'selected="selected"' : '' }
+              data-url="${ version.urls.documentation }">
+              ${ version.slug }
+          </option>`
+      ).join("\n") }
+    </select>
+  `;
+
+  dropdownSelectors.insertAdjacentHTML("beforeend", versionSelect);
+  const versionSelector = document.querySelector("#readthedocs-version-selector");
+  versionSelector.addEventListener("change", onSelectorSwitch);
+  versionSelector.innerHTML = versionSelect;
```

Comment:
I'm having a bit of trouble figuring out why some of this is needed. It feels like adding, querying, then adding the elements again could be simplified into something easier to follow.

The reason I hardcoded multiple inner divs in `div.switch-menus` is so that we already have an easy target for this HTML:

```suggestion
  versionSwitch.innerHTML = versionSelect;
  versionSwitch.firstChild.addEventListener("change", onSelectorSwitch);
```

Reply:
I followed this approach and it worked fine.